Dockerfile.buildx

```diff
@@ -10,5 +10,10 @@ ARG GOOS
 ARG GOARCH
 # Keep these essentially no-op var settings for debug purposes.
 # It allows us to see what the GOOS/GOARCH that's being built for is.
-RUN GOOS=${GOOS} GOARCH=${GOARCH} go build -ldflags '-X github.com/docker/buildx/version.Version=${BUILDX_COMMIT} -X github.com/docker/buildx/version.Revision=${BUILDX_COMMIT} -X github.com/docker/buildx/version.Package=github.com/docker/buildx' -o /usr/bin/buildx ./cmd/buildx
-ENTRYPOINT ["/usr/bin/buildx"]
\ No newline at end of file
+RUN GOOS="${GOOS}" GOARCH="${GOARCH}" BUILDX_COMMIT="${BUILDX_COMMIT}" \
+    ldflags="\
+    -X \"github.com/docker/buildx/version.Version=$(git describe --tags)\" \
+    -X \"github.com/docker/buildx/version.Revision=$(git rev-parse --short HEAD)\" \
+    -X \"github.com/docker/buildx/version.Package=github.com/docker/buildx\"" \
+ && go build -ldflags "${ldflags}" -o /usr/bin/buildx ./cmd/buildx
```

Comment:
`&&` looks artificial here as we don't have any command to succeed before it.

I also see a repetition of `github.com/docker/buildx` string; why don't we do this

```shell
  pkg="github.com/docker/buildx"; \
  ldflags="\
    -X \"${pkg}/version.Version=$(git describe --tags)\" \
    -X \"${pkg}/version.Revision=$(git rev-parse --short HEAD)\" \
    -X \"${pkg}/version.Package=${pkg}\""; \
  echo go build -ldflags "${ldflags}" -o /usr/bin/buildx ./cmd/buildx
```

I checked it works in both `bash` and `dash`, and even passes shellcheck

Reply:
> && looks artificial here as we don't have any command to succeed before it.

arf. that's me messing up rebasing

> I also see a repetition of github.com/docker/buildx string; why don't we do this

👍 yup, I like that